sdk/core/azure-core-amqp/src/test/java/com/azure/core/amqp/implementation/TransactionCoordinatorTest.java

```diff
@@ -0,0 +1,122 @@
+
+        TransactionCoordinator transactionCoordinator = new TransactionCoordinator(sendLink, messageSerializer);
+
+        doReturn(Mono.just(outcome)).when(sendLink).send(any(byte[].class), anyInt(), eq(DeliveryImpl.DEFAULT_MESSAGE_FORMAT), isNull());
+
+        StepVerifier.create(transactionCoordinator.completeTransaction(transaction, isCommit))
+            .verifyError(IllegalArgumentException.class);
+
+        verify(sendLink, times(1)).send(any(byte[].class), anyInt(), eq(DeliveryImpl.DEFAULT_MESSAGE_FORMAT), isNull());
+    }
+
+    @MethodSource("commitParams")
+    @ParameterizedTest
+    public void testCompleteTransaction(boolean isCommit) {
+        final Accepted outcome = Accepted.getInstance();
+
+        final AmqpTransaction transaction = new AmqpTransaction(ByteBuffer.wrap("1".getBytes()));
+
+        TransactionCoordinator transactionCoordinator = new TransactionCoordinator(sendLink, messageSerializer);
+
+        doReturn(Mono.just(outcome)).when(sendLink).send(any(byte[].class), anyInt(), eq(DeliveryImpl.DEFAULT_MESSAGE_FORMAT), isNull());
+
+        StepVerifier.create(transactionCoordinator.completeTransaction(transaction, isCommit))
+            .verifyComplete();
+
+        verify(sendLink, times(1)).send(any(byte[].class), anyInt(), eq(DeliveryImpl.DEFAULT_MESSAGE_FORMAT), isNull());
+    }
+
+    @Test
+    public void testCreateTransactionRejected() {
+        Rejected outcome = new Rejected();
+
+        final TransactionCoordinator transactionCoordinator = new TransactionCoordinator(sendLink, messageSerializer);
+
+        doReturn(Mono.just(outcome)).when(sendLink).send(any(byte[].class), anyInt(), eq(DeliveryImpl.DEFAULT_MESSAGE_FORMAT), isNull());
+
+        StepVerifier.create(transactionCoordinator.createTransaction())
+            .verifyError(IllegalArgumentException.class);
+
+        verify(sendLink, times(1)).send(any(byte[].class), anyInt(), eq(DeliveryImpl.DEFAULT_MESSAGE_FORMAT), isNull());
+    }
+
+    @Test
+    public void testCreateTransaction() {
+        final Duration shortTimeout = Duration.ofSeconds(5);
+        final byte[] transactionId = "1".getBytes();
+        Declared transactionState = new Declared();
+        transactionState.setTxnId(Binary.create(ByteBuffer.wrap(transactionId)));
+
+        TransactionCoordinator transactionCoordinator = new TransactionCoordinator(sendLink, messageSerializer);
+
+        doReturn(Mono.just(transactionState)).when(sendLink).send(any(byte[].class), anyInt(), anyInt(), isNull());
+
+        AmqpTransaction actual = transactionCoordinator.createTransaction().block(shortTimeout);
+
+        Assertions.assertNotNull(actual);
+        Assertions.assertArrayEquals(transactionId, actual.getTransactionId().array());
+        verify(sendLink, times(1)).send(any(byte[].class), anyInt(), eq(DeliveryImpl.DEFAULT_MESSAGE_FORMAT), isNull());
+    }
+
+    protected static Stream<Arguments> commitParams() {
```

Comment:
* Why protected? 
* There is a ValueSource that supports boolean. https://junit.org/junit5/docs/current/api/org.junit.jupiter.params/org/junit/jupiter/params/provider/ValueSource.html
* Params -> Parameters.